src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java

```diff
@@ -108,4 +109,14 @@
     @OneToMany(mappedBy = "discount")
     @EqualsAndHashCode.Exclude
     private List<UserDiscount> users = new ArrayList<>();
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToMany
+    @JoinTable(
+            name = "favorites",
+            joinColumns = @JoinColumn(name = "dis_id"),
+            inverseJoinColumns = @JoinColumn(name = "usr_id")
+    )
+    private List<User> favoriteUsers = new ArrayList<>();
```

Comment:
as for me the variable with name userFavorites would be more suitable here)

Reply:
changed